src/datasets/arrow_writer.py

```diff
@@ -456,7 +456,11 @@ def check_duplicate_keys(self):
         tmp_record = set()
         for hash, key in self.hkey_record:
             if hash in tmp_record:
-                raise DuplicatedKeysError(key)
+                duplicate_key_indices = [
+                    str(index) for index, (duplicate_hash, _) in enumerate(self.hkey_record) if duplicate_hash == hash
```

Comment:
I think this is what @mariosasko meant:

```suggestion
                    str(self._num_examples  + index) for index, (duplicate_hash, _) in enumerate(self.hkey_record) if duplicate_hash == hash
```